src/software/backend/simulation/box2d_util.cpp

```diff
@@ -1,8 +1,8 @@
 #include "software/backend/simulation/box2d_util.h"
 
-bool bodyExistsInWorld(b2Body* body, std::shared_ptr<b2World> world)
+bool bodyExistsInWorld(b2Body* body, b2World* world)
```

Comment:
seconded :nauseated_face:

Reply:
I knew I had a reason but forgot, here's the explanation.

Basically, b2Body objects should be destroyed by the world. Their destructor is marked `private` to force this to happen. 
https://github.com/erincatto/Box2D/blob/master/Box2D/Dynamics/b2Body.h
https://github.com/erincatto/Box2D/blob/master/Box2D/Dynamics/b2Body.cpp

This means we can't even create a `shared_ptr` from the `b2Body*` returned when we create a body like `std::shared_ptr<b2Body>(world->CreateBody(&body_def))`. This fails because the smart pointer template will try make a call to the private destructor.

So we can't actually use smart pointers as member variables to store the bodies in our classes, so the utility functions have to handle raw pointers. Unfortunately, using smart pointers also doesn't save us the trouble of implementing our own destructor because of how bodies have to be destroyed using the b2World.

Reply:
Happy to document this somewhere. Open to ideas on where I can put this that won't need to be copied / duplicated all the time (like in each abstraction class), but is relatively visible.

Reply:
If the compiler won't allow it to happen then I'm fine with it as-is honestly.